app/assets/javascripts/components/shared/ModalDialog.tsx

```diff
@@ -1,19 +1,9 @@
-import { ComponentChildren, FunctionComponent } from 'preact';
```

Comment:
Yes, `components` folder is where we put our React components. As of reusable components (like this modal), I prefer combining them in a separate folder (like "shared"), which gives at least 2 advantages:
1. we know where to look for any reusable component
2. the `components` folder itself doesn't become cluttered and contains only "unique" components

Reply:
But all of the components in the components folder are meant to be reusable and not tightly coupled with their context, no?

Reply:
No, for example `HelpAndFeedback` or `PasscodeLock` are related to "Preferences -> Help" or "Account -> Passcode lock", thus they are specific to their respective sections. But Modal is intended to be used from anywhere, that makes it really reusable (or shared) component. That's what I mean by saying reusable components.

Reply:
That's why HelpFeedback is in the preferences folder, not the components folder. Don't know about PasscodeLock.

Reply:
In `components` folder there are a lot of components already, and quite a few of them are specific, like ("NotesOptionsPanel", "SessionsModal", etc.). Putting shared components there can clutter that folder pretty quickly, thus putting reusable components into some "shared" folder will help having more organized structure. And it will be easier to find them knowing they are all in "shared" folder (although there are other files that would be better to move to "shared" folder as well, but not in this PR :) ).